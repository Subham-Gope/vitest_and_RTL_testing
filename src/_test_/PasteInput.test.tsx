// import { describe, test, expect } from "vitest";
// import { screen, render } from "@testing-library/react";
// import userEvent from "@testing-library/user-event";
// import PasteInput from "../components/PasteInput";

// describe("Testing the PasteInput.tsx file", () => {
//   test("displays the pasted text", async () => {
//     // setting up the userEvent
//     const user = userEvent.setup();

//     // setting up the debugger
//     screen.debug();

//     // setting up the snapshot matching
//     const { container } = render(<PasteInput />);
//     expect(container).toMatchSnapshot();

//     const input = screen.getByRole("textbox", {
//       name: /paste your text here./i,
//     });
//     expect(input).toBeInTheDocument();

//     // Initiating user interaction
//     const text = "Hello";
//     input.focus();
//     await user.paste(text);
//     expect(input).toHaveValue(text);
//   });
// });
