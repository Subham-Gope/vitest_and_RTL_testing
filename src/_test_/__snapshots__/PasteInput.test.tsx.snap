// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing the PasteInput.tsx file > displays the pasted text 1`] = `
<div>
  <div>
    <label
      for="paste"
    >
      Paste Your Text here.
    </label>
    <input
      id="paste"
      name="paste"
      placeholder="paste here"
      type="text"
      value=""
    />
  </div>
</div>
`;
